### PR TITLE
New Rule: maximumNumberOfLines

### LINKED ISSUE #1181: New rule: maximumNumberOfLines
Since I really dislike files with several thousand lines of code, I'd like to have a rule in which I can specify the maximal number of lines a file can have. 

Maybe this rule should have an option to not count empty lines or comments. 

Any opinions?

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/9749988-new-rule-maximumnumberoflines?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by hzoo**
Fixes #1181. @alawatthe

Just a start - didn't add any options yet for newlines or comments.

### CODE REVIEW

**Comment by alawatthe** [comment]
Just repeating this (since my inline comment in the first commit somehow vanished):
Everything in the pull request looks good to me. I would be happy to see it merged.

**Comment by hzoo** [comment]
I think it was because you commented on the commit that I rebased and I guess it got removed... yeah that's still weird though.

File: lib/config/configuration.js [hunk 1]
```diff
@@ -684,6 +684,8 @@ Configuration.prototype.registerDefaultRules = function() {
     this.registerRule(require('../rules/disallow-keywords-in-comments'));
 
     this.registerRule(require('../rules/disallow-identifier-names'));
+
+    this.registerRule(require('../rules/maximum-number-of-lines'));
```
Comment (zxqfox) [inline lib/config/configuration.js]: I think we can move this to `maximum-line-length`

**Comment by zxqfox** [comment]
:+1: to merge

**Pull request closed without merging**